### PR TITLE
add --no-wait to az group delete in delete.sh

### LINKED ISSUE #501: Consider allowing --no-wait via hack/delete.sh
Should speed up our CI jobs

On a tangent, we may want to think whether it's possible to postpone GC of resource groups to be run only via purge to enable reuse of resource groups when retesting the same commit and the previous run failed during the e2e test (to avoid the extra time of reinstalling a cluster).

### PR DESCRIPTION
**Pull request by charlesakalugwu**
closes #501 

I also tested what happens if create.sh is run right after.

```
The resource group 'charlesakalugwu' is in deprovisioning state and cannot perform this operation.
```
So we are ok.

### CODE REVIEW

**Comment by kargakis** [comment]
I am not sure I would run this by default. In the dev case, it probably makes sense to wait for the deletion to complete, when you want to recreate the same resource group. I'd probably enable this via an env variable. @jim-minter thoughts?

**Comment by mjudeikis** [comment]
Not for every case. In ci we for sure want to wait in CI jobs.

**Comment by kargakis** [comment]
> Not for every case. In ci we for sure want to wait in CI jobs.

Depends on how we name our resource groups. In most tests, I think the resource group depends on the namespace so we may want to ensure RGs in those tests get a unique name if we want to proceed with this.

**Comment by charlesakalugwu** [comment]
you can now specify `NO_WAIT=true` to enable `--no-wait` on az group delete

File: hack/delete.sh [hunk 1]
```diff
@@ -14,6 +14,13 @@ if [[ -z "$DNS_RESOURCEGROUP" ]]; then
     exit 1
 fi
 
+NO_WAIT=${NO_WAIT:false}
```
[the remark this answers is not in the document]
Comment (charlesakalugwu) [reply hack/delete.sh]: sure

File: hack/delete.sh [hunk 2]
```diff
@@ -25,4 +32,4 @@ hack/dns.sh zone-delete $RESOURCEGROUP
 
 rm -rf _data
 
-az group delete -n $RESOURCEGROUP -y
+az group delete -n $RESOURCEGROUP -y ${DELETE_FLAGS}
```
[the remark this answers is not in the document]
Comment (charlesakalugwu) [reply hack/delete.sh]: done

**Comment by kargakis** [comment]
Please squash down to a single commit

**Comment by jim-minter** [comment]
manually squash merged

**Pull request merged**